Civi/Api4/Utils/CoreUtil.php

```diff
@@ -244,26 +244,24 @@ public static function checkAccessRecord(AbstractAction $apiRequest, array $reco
 
     // For get actions, just run a get and ACLs will be applied to the query.
     // It's a cheap trick and not as efficient as not running the query at all,
-    // but BAO::checkAccess doesn't consistently check permissions for the "get" action.
+    // but authorizeRecord doesn't consistently check permissions for the "get" action.
     if (is_a($apiRequest, '\Civi\Api4\Generic\AbstractGetAction')) {
       return (bool) $apiRequest->addSelect($idField)->addWhere($idField, '=', $record[$idField])->execute()->count();
     }
 
     $event = new \Civi\Api4\Event\AuthorizeRecordEvent($apiRequest, $record, $userID);
     \Civi::dispatcher()->dispatch('civi.api4.authorizeRecord', $event);
 
-    // Note: $bao::_checkAccess() is a quasi-listener. TODO: Convert to straight-up listener.
+    // $bao::_checkAccess() is deprecated in favor of `civi.api4.authorizeRecord` event.
```

Comment:
I don't think so, because there weren't really any listeners out there as this code is all quite new. But one of the advantages to switching to an event callback is that now you *can* control the order, by using `EventSubscriberInterface` and setting the priority of your callback.